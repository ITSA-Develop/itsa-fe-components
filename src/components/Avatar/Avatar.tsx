import { Avatar as AntAvatar, AvatarProps } from 'antd';

export interface IAvatarProps extends AvatarProps {
	name?: string;
}
const stringAvatar = (name: string): string => {
	const splittedFullname = name.trim().split(' ').filter(Boolean);

	const firstLetter = splittedFullname[0]?.[0] ?? '';
	const secondLetter = splittedFullname[1]?.[0] ?? '';

	return firstLetter + secondLetter || firstLetter;
};

export const Avatar = ({ name, ...rest }: IAvatarProps) => {
	return <AntAvatar {...rest}>{stringAvatar(name || '') || ''}</AntAvatar>;
};
