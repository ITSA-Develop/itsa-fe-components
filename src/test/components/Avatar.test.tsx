import { render, screen } from '@testing-library/react';
import { beforeAll, describe, expect, it } from 'vitest';
import { Avatar } from '../../components/Avatar/Avatar';

beforeAll(() => {
	Object.defineProperty(window, 'matchMedia', {
		writable: true,
		value: (query: string) => ({
			matches: false,
			media: query,
			onchange: null,
			addListener: () => {},
			removeListener: () => {},
			addEventListener: () => {},
			removeEventListener: () => {},
			dispatchEvent: () => false,
		}),
	});
});

describe('Avatar component', () => {
	it('shows initials for two-word name', () => {
		const { container } = render(<Avatar name="John Doe" />);
		expect(screen.getByText('JD')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('shows initial for single-word name', () => {
		const { container } = render(<Avatar name="Alice" />);
		expect(screen.getByText('A')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('renders nothing if name is not provided', () => {
		const { container } = render(<Avatar />);
		expect(screen.queryByText(/./)).toBeNull();
		expect(container).toMatchSnapshot();
	});

	it('passes additional props to AntAvatar', () => {
		const { container } = render(<Avatar name="John Doe" size={64} />);
		const avatar = screen.getByText('JD').parentElement;
		expect(avatar).toBeDefined();
		expect(avatar).toHaveStyle({ width: '64px', height: '64px' });
		expect(container).toMatchSnapshot();
	});
});
