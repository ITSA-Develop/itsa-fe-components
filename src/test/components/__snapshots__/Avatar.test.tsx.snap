// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar component > passes additional props to AntAvatar 1`] = `
<div>
  <span
    class="ant-avatar ant-avatar-circle css-dev-only-do-not-override-l9pxc0"
    style="width: 64px; height: 64px; font-size: 18px;"
  >
    <span
      class="ant-avatar-string"
      style="-webkit-transform: scale(1); transform: scale(1);"
    >
      JD
    </span>
  </span>
</div>
`;

exports[`Avatar component > renders nothing if name is not provided 1`] = `
<div>
  <span
    class="ant-avatar ant-avatar-circle css-dev-only-do-not-override-l9pxc0"
  >
    <span
      class="ant-avatar-string"
      style="-webkit-transform: scale(1); transform: scale(1);"
    />
  </span>
</div>
`;

exports[`Avatar component > shows initial for single-word name 1`] = `
<div>
  <span
    class="ant-avatar ant-avatar-circle css-dev-only-do-not-override-l9pxc0"
  >
    <span
      class="ant-avatar-string"
      style="-webkit-transform: scale(1); transform: scale(1);"
    >
      A
    </span>
  </span>
</div>
`;

exports[`Avatar component > shows initials for two-word name 1`] = `
<div>
  <span
    class="ant-avatar ant-avatar-circle css-dev-only-do-not-override-l9pxc0"
  >
    <span
      class="ant-avatar-string"
      style="-webkit-transform: scale(1); transform: scale(1);"
    >
      JD
    </span>
  </span>
</div>
`;
